### ISSUE TITLE
Better use of abbreviated unit identifiers

### ISSUE THREAD

**Issue by JohelEGP** [nholthaus/units#196]
For a long while now, I've thought that it'd be great if we could naturally write the following.
```C++
using namespace units::literals;
using namespace units::constants;
const auto speed_limit = 90_km/h;
```
However, we need to write `90_km/1_h` as the abbreviated unit identifiers happened to be taken up by some "tags."

What do you think about repurposing these identifiers to somewhere else, like the `units::constants` namespace, where they'd be declared along the lines of:
```C++
inline constexpr time::hour_t<int> h(1);
```
This would allow us to more naturally express units in our programs. The standard time library already allows expressing dates using the natural syntax, like [`1970y/January/1`](http://eel.is/c++draft/time#clock.system.overview-1.example-1). I don't see why we shouldn't follow suit.

**Comment by nholthaus** [comment]
Yes, this. It never crossed my mind honestly but I _love_ the natural syntax. That's the ethos and purpose of the library in the first place.

**Comment by johelegp** [comment]
I had thought about it, and people `using namespace units::literals;` for the UDLs might find themselves having undesired clashes or accidentally using these kind of "constant literals". However, I haven't come up with something better.

**Comment by nholthaus** [comment]
hmm, good point. I still like inlining the real constants namespace, we can
call these `literal_constants` or something like that. That said, I may be
OK leaving the clash resolution as an exercise to the user. If you have a
bunch of single letter variables or unit abbreviations as variables, you
may need/want a refactor for units anyway.

**Comment by johelegp** [comment]
Yeah. `literals` might be good enough most of the time.

**Comment by nholthaus** [comment]
if not we'll just patch it out in v3.1 I suppose...

**Comment by johelegp** [comment]
#256 removes the abbreviation-named tags.

**Comment by johelegp** [comment]
#212 blocks a general solution for this.

**Comment by nholthaus** [comment]
all the concentration unit names were expanded, so only dB uses an abbreviation. I'm pretty sure we don't want to use operators like `/` and `*` to change linear scales, since arithmetic has a different meaning in log space.

**Comment by johelegp** [comment]
Alternatively, `dB` should be pluralized, like how all other units were renamed (I don't know if there are others). Then `dB` would be freed up for this.

**Comment by nholthaus** [comment]
I don't have a problem with pluralizing the names, but I don't think it will work as intended. I can't do `20 * dB * W` because `operator*` isn't defined for `decibel_scale` types, and I think even if it did that would still result in `watts`?

**Comment by johelegp** [comment]
Right. Let's leave non-linear scales off this.

**Comment by nholthaus** [comment]
I think we're good to go then, I really want this as part of the release.

**Comment by johelegp** [comment]
Well, it should be easy to implement, given that only arithmetic types are supported at the moment.

**Comment by johelegp** [comment]
I have an initial implementation: https://github.com/johelegp/units/tree/natural_syntax. But the restriction of the underlying type being an arithmetic type gets in the way. It'd be easier if there was support for generalized underlying types.

**Comment by nholthaus** [comment]
totally agree. I think the criticisms against being able to use `std::complex` are particularly valid. 

I don't think there was any additional rationale for `arithmetic` underlying types than that it was obvious and a trait was readily available. I might have thought it was advantageous for `trivial` units, but I forget the motivation behind that. Probably some `constexpr` limitation in older compilers.

**Comment by nholthaus** [comment]
Abbreviation-named unit constants are provided for every unit (UNIT_ADD_CONSTANT, include/units/core.h), so natural syntax such as `90.0 * km / hr` composes directly into a kilometers_per_hour velocity. The abbreviation-named tags that previously blocked this were removed; abbreviations are now unit constants. Note that `h` denotes the Planck constant, so hour uses the `hr` constant. Closing as implemented.
